src/decode.h

```diff
@@ -457,7 +457,7 @@ typedef struct Packet_
      * has the exact same tuple as the lower levels */
     uint8_t recursion_level;
 
-    uint16_t vlan_id[2];
+    uint16_t vlan_id[3];
```

Comment:
I think Victor wanted to avoid extending this structure, and do something clever with all optional fields, such as MPLS, vlan et al...